prometheus/src/main/java/org/opensearch/sql/prometheus/client/PrometheusClient.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.prometheus.client;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.io.IOException;
+import java.util.List;
+import org.json.JSONObject;
+
+public interface PrometheusClient {
+
+  JSONObject queryRange(String query, Long start, Long end, String step) throws IOException;
+
+  String[] getLabels(String metricName) throws IOException;
+
+  void schedule(Runnable task);
```

Comment:
Does schedule function required? Prometheus query execute in sql-worker thread, right?

Reply:
Not required. I think Execution Engine is using OpensearchClient Schedule function to run the task.